### DISCUSSION TITLE
wtpy SEL引擎回测股票时，stra_get_time 返回都是 2359，而且context.stra_get_bars 获取到的数据为空

### DISCUSSION THREAD

**Discussion by RickyLoveFreedom** [wondertrader/wondertrader#57] [Q&A] [answered]
分支版本:dev-0.93
ExtDataLoader 获取数据用的postgresql数据库中准备好的数据，并且是能正常加载数据的。
![image](https://user-images.githubusercontent.com/21985820/162354592-01f9c44b-5460-458c-b875-ee7d7adc9fad.png)

策略模板是用的 demos/sel_fut_bt/Strategies/DualThrust_Sel.py 这个模板，结果回测时，isInTradingTime(curTime) 一直返回False,因为curTime = 2359，导致回测无法执行，而且当注释这个交易时间的判断后，context.stra_get_bars 返回数据为空
![image](https://user-images.githubusercontent.com/21985820/162355431-2bcae077-87dd-4ba4-aa5f-1ee19413d88b.png)

是不是我哪里的配置有问题呢 ?

另外请教下 extDataLoader能否增加一个自定义的dict参数,用于传入一些自定义参数，来限制每次获取数据的条数，因为现在看到每次查询查询标的时，都是查询该标的所有的数据，比如我只想回测 2020年-2021年时间段内的数据，但是因为在extDataLoader中获取不到回测区间参数，每次默认查询该标的  1991年到2022年时间段内的数据，这样对查询性能比较低。

**Comment by wondertrader** [comment, marked as answer]
可否看一下你的runBT.py里怎么写的？

**Reply by RickyLoveFreedom** [reply]
> 可否看一下你的runBT.py里怎么写的？
[test_stra.zip](https://github.com/wondertrader/wondertrader/files/8458130/test_stra.zip)
**不过我是测试的  testDtLoader.py 这个启动类**

**Reply by wondertrader** [reply]
```python
from wtpy import WtBtEngine,EngineType

from Strategies.DualThrust_Sel import StraDualThrustSel

# from Strategies.XIM import XIM

if __name__ == "__main__":
    #创建一个运行环境，并加入策略
    engine = WtBtEngine(EngineType.ET_SEL)
    engine.init('../common/', "configbt.yaml")
    engine.configBacktest(201909100900,202008071500)
    engine.configBTStorage(mode="csv", path="../storage/")
    engine.commitBTConfig()

    straInfo = StraDualThrustSel(name='DT_COMM_SEL', codes=["CFFEX.IF.HOT","SHFE.rb.HOT","DCE.i.HOT"], barCnt=50, period="m5", days=30, k1=0.1, k2=0.1)
    engine.set_sel_strategy(straInfo, time=5, period="day")

    engine.run_backtest()

    kw = input('press any key to exit\n')
    engine.release_backtest()
```
你把下面这一行
```python
engine.set_sel_strategy(straInfo, time=5, period="day")
```
中的*period*改成*d*试试看

**Reply by RickyLoveFreedom** [reply]
![image](https://user-images.githubusercontent.com/21985820/162861115-ab98ef1a-0f0e-4f16-8080-562ab05c99e3.png)
改了  还是一样的获取不到数据

**Reply by wondertrader** [reply]
![image](https://user-images.githubusercontent.com/62869597/162862575-b02bfdac-bb70-4df5-a3ad-8f34c3065626.png)

**Reply by RickyLoveFreedom** [reply]
我的错，在数据落地的时候，有部分数据被过滤掉了，辛苦老大了。
还有SEL回测 context.stra_get_time() 不在交易时间内的问题，也麻烦看下
extDataLoader能不能添加自定义参数，传入回测区间，来减少查询数据量，提升一点性能呢 ？
